pints/_mcmc/_result.py

```diff
@@ -0,0 +1,157 @@
+#
+# MCMC result object
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+# Some code in this file was adapted from Myokit (see http://myokit.org)
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints._diagnostics as diagnostics
+import numpy as np
+from tabulate import tabulate
+
+
+class McmcResultObject(object):
+    """
+    Class that represents the result of running MCMC and
+    calculates key parameter summaries which are printed to the
+    console, including the posterior mean, standard deviation,
+    quantiles, rhat and effective sample size
+    """
+
+    def __init__(self, chains, time):
+        self._chains = chains
+        stacked = np.vstack(chains)
+        self._time = time
+        self._mean = np.mean(stacked, axis=0)
+        self._std = np.std(stacked, axis=0)
+        self._quantiles = np.percentile(stacked, [2.5, 25, 50,
+                                                        75, 97.5], axis=0)
+        self._ess = diagnostics.effective_sample_size(stacked)
```

Comment:
would it make sense to move this calculation to the make_summary() method? So if the user doesn't want the ess it isn't calculated.